### PR TITLE
ref(ts): convert pluginActions to TS

### PR DESCRIPTION
**Pull request by scefali**

### CODE REVIEW

File: src/sentry/static/sentry/app/components/group/pluginActions.tsx [hunk 1]
```diff
@@ -112,7 +129,7 @@ class PluginActions extends React.Component {
           enforceFocus={false}
         >
           <Modal.Header closeButton>
-            <Modal.Title>{`${plugin.name || plugin.title} Issue`}</Modal.Title>
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/group/pluginActions.tsx:115]: Are we sure that `name` always exists and is not an empty string?

File: src/sentry/static/sentry/app/components/group/pluginActions.tsx [hunk 2]
```diff
@@ -87,7 +104,7 @@ class PluginActions extends React.Component {
     });
   };
 
-  handleClick = actionType => {
+  handleClick = (actionType: 'create' | 'link') => {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/group/pluginActions.tsx:107]: Can this be `Exclude<State['actionType'], null>`?

File: src/sentry/static/sentry/app/components/group/pluginActions.tsx [hunk 3]
```diff
@@ -10,17 +11,33 @@ import {t} from 'app/locale';
...
+  api: Client;
+  group: Group;
+  organization: Organization;
+  project: Project;
+  plugin: Plugin;
+};
+
+type State = {
+  showModal: boolean;
+  actionType: 'create' | 'link' | null;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/group/pluginActions.tsx:26]: I would create a separate type for this, like for example:

type Action = 'create' | 'link'

if you do so you can reuse it in the `handleClick` function 😉

File: src/sentry/static/sentry/app/components/group/pluginActions.tsx [hunk 4]
```diff
@@ -112,7 +129,7 @@ class PluginActions extends React.Component {
           enforceFocus={false}
         >
           <Modal.Header closeButton>
-            <Modal.Title>{`${plugin.name || plugin.title} Issue`}</Modal.Title>
+            <Modal.Title>{`${plugin.name} Issue`}</Modal.Title>
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/group/pluginActions.tsx:132]: I believe you should use translation here

File: src/sentry/static/sentry/app/components/group/pluginActions.tsx (hunk 1 above)
Comment (scefali) [reply src/sentry/static/sentry/app/components/group/pluginActions.tsx:115]: @billyvg It should be but I was able to keep this by fixing the type

File: src/sentry/static/sentry/app/components/group/pluginActions.tsx (hunk 3 above)
Comment (scefali) [reply src/sentry/static/sentry/app/components/group/pluginActions.tsx:29]: Went with Billy's suggestion to handle this

File: src/sentry/static/sentry/app/components/group/pluginActions.tsx [hunk 5]
```diff
@@ -2,25 +2,45 @@ import PropTypes from 'prop-types';
...
-class PluginActions extends React.Component {
-  static propTypes = {
+type Props = {
+  api: Client;
+  group: Group;
+  organization: Organization;
+  project: Project;
+  plugin: Plugin & {
+    // issue serializer adds more fields
+    title: 'string';
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/group/pluginActions.tsx:23]: Can we add a todo note here that this shouldn't be needed as we think `name` should always exist, but it's here for compatibility reasons? I wonder if it'd be worthwhile to use `trackDeprecated` to see if this is being hit at all (`name` being empty) and just removing it if so.

```suggestion
    title: string;
```

**Review by billyvg — APPROVED** [review]
Looks good besides the type thing.

**Pull request merged**